p2p/agent.go

```diff
@@ -187,6 +188,10 @@ func (p *Agent) Start(ctx context.Context) error {
 			skip = true
 			return
 		}
+		if broadcast.ChainId != p.cfg.ChainID {
+			err = errors.Errorf("chain ID mismatch, received %d, expecting %d", broadcast.ChainId, p.cfg.ChainID)
+			return
+		}
```

Comment:
also add this check to protect possible mainnet/testnet mix-up
in case a mainnet node joins testnet or vice versa, this would prevent node processing wrong msg (from other network)